packages/playwright-core/src/web/traceViewer/ui/workbench.tsx

```diff
@@ -65,6 +68,13 @@ export const Workbench: React.FunctionComponent<{
     processTraceFile(event.target.files[0]);
   };
 
+  React.useEffect(() => {
+    const newTraceURL = new URL(window.location.href).searchParams.get('trace');
+    // Don't re-use blob file URLs on page load (results in Fetch error)
+    if (newTraceURL && !newTraceURL.startsWith('blob:'))
+      setTraceURL(newTraceURL);
+  }, [setTraceURL]);
```

Comment:
What does it mean useEffect around setTraceURL, wouldn't it run only once?

Reply:
Yes, as the comment says its about if you open trace viewer with a blob URL. Usually old tab, forced reload etc.